### ISSUE TITLE
Upgrading project dependencies

### ISSUE THREAD

**Issue by T-bond** [domoticz/domoticz-android#499]
I am currently tried to implement the new Fingerprint dialog from the support library, when I figured out, that the support librarys latest version [was](https://developer.android.com/topic/libraries/support-library/revisions) the 28.0.0.
So I updated all the dependencies to this new AndroidX support library. I would like to remove the Digitus dependency, and use the AndroidX one's, as it will make it more consistent with the default one, and will support all the biometric options.
Also with the updated version we get all the sweet bugfixes.

During this process I fix all the warnings I can find, which was there already, or got there during the process. (like duplicated entries in the manifest, or not used ones)

Already made the upgrade on my local repository, and I can build it and it seems to be working, but I think it will need a deeper testing from our part, if you accept this change / PR. I am only testing the premium version (also bought on GPlay), and feeling a little bit rusty in terms of android development as I haven't touched Android in a few years, so maybe it won't be the best transition, but I am trying my best.

@galadril my main question is, would like to see a PR request with the above mentioned changes, and if so, would you so kind (and maybe the community) to test it after that, as this procedure seems to be a too big task for only one people.

Thanks,
    T-bond

**Comment by galadril** [comment]
Yeah you can submit a PR, ill test it with different devices / domoticz setups.

**Comment by T-bond** [comment]
This PR is already getting really big. (You can check the progress [here](https://github.com/T-bond/domoticz-android/tree/dependencies-update) if you interested)
It is in a very WIP state, but I am working my way through.

I am also migrating away from Material Dialogs library (with Digitus too) to the AndroidX DialogAlert as MD lib got rewritten from the ground up, and it seems like they are now using Kotlin and it doesn't play nice with Java. The only advantage would be the InputDialog but id could be easily done with a custom view.

On the way, I try to make small improvements, like replacing `x.size() > 1` to `!x.isEmpty()` and making the code a bit more manageable.

**Comment by galadril** [comment]
Ok sounds cool, ill keep an eye out for your progress.
Do you also have any experience with widgets? People are having issues that widgets are invalid after app updates, but I can't really find the issue here

**Comment by galadril** [comment]
See my latest alpha checkins, updated the dependencies to androidx

**Comment by T-bond** [comment]
Nice. Than I will only need to create a pull request for the Digitus and Material Dialog removal.
I will need a few days, as I am currently having some exams, but I will have the last tomorrow, so I can continue on the develop.

**Comment by galadril** [comment]
https://github.com/domoticz/domoticz-android/issues/511

**Comment by T-bond** [comment]
Yes. I was ready with the PR but unfortunately didn't have the time, to polish and submit it. But now maybe I can improve some other things in the app.
